### PR TITLE
Handle UCM content and not nullable datetime columns, too, and do other fixes from code reviews

### PR DESCRIPTION
**Pull request by richard67**
Pull Request for [https://github.com/joomla/joomla-cms/pull/26295](https://github.com/joomla/joomla-cms/pull/26295).

### Summary of Changes

Handle UCM content for articles, too, and apply a few fixes from @Quy 's code review on the CMS PR mentioned above.

Add update script "4.0.0-2018-08-01.sql" for UCM content which comes early enough in the chain for all later PR's being able to use null values for datetimes, too, like it is added to this PR for com content articles.

Add handling of not nullable datetime columns as discussed yesterday on Glip.

### CODE REVIEW

File: libraries/src/Table/CoreContent.php [hunk 1]
```diff
@@ -271,6 +279,11 @@ public function store($updateNulls = false)
 				$this->core_created_user_id = $user->get('id');
 			}
 
+			if (!(int) $this->core_modified_time)
+			{
+				$this->core_modified_time = this->core_created_time;
```
Comment (Quy) [inline libraries/src/Table/CoreContent.php]:
```suggestion
				$this->core_modified_time = $this->core_created_time;
```

File: components/com_content/Model/ArticlesModel.php [hunk 2]
```diff
@@ -484,8 +481,8 @@ function ($data) use ($db)
 		switch ($dateFiltering)
 		{
 			case 'range':
-				$startDateRange = $db->quote($this->getState('filter.start_date_range', $nullDate));
-				$endDateRange   = $db->quote($this->getState('filter.end_date_range', $nullDate));
+				$startDateRange = $db->quote($this->getState('filter.start_date_range'));
```
Comment (richard67) [inline components/com_content/Model/ArticlesModel.php:484]: @wilsonge Here and in the following line you have to advise me if that is ok or if there has to be something done for one of the range start and end dates being not specified, or both even. The existing code might do the wrong thing in such case, if that case is allowed.

Comment (wilsonge) [reply components/com_content/Model/ArticlesModel.php:484]: what are the options for this? We've effectively mandated `a.created` and `a.modified`

Comment (richard67) [reply components/com_content/Model/ArticlesModel.php:484]: no, these here are start and end dates for a range filtering, i think. they are entered in a filter. the date to be fitered for is either created or modified, but my question above refers to that start and end dates of the range for filtering. or am i getting something wrong?

Comment (wilsonge) [reply components/com_content/Model/ArticlesModel.php:484]: ok can be `a.publish_up` (mod_articles_popular and mod_articles_category). Just need to test this still works

**Pull request merged**